coderd/database/queries/workspaces.sql

```diff
@@ -161,17 +178,41 @@ WHERE
 			name ILIKE '%' || @name || '%'
 		ELSE true
 	END
+	-- Filter by agent status
+	-- has-agent: is only applicable for workspaces in "start" transition. Stopped and deleted workspaces don't have agents.
+	AND CASE
+		WHEN @has_agent :: text != '' THEN
+			latest_build.transition = 'start'::workspace_transition
+			AND CASE
+				WHEN @has_agent = 'timeout' THEN
+					latest_build.first_connected_at IS NULL AND (latest_build.created_at + latest_build.connection_timeout_seconds * INTERVAL '1 second' < NOW())
+				WHEN @has_agent = 'connecting' THEN
+					latest_build.first_connected_at IS NULL
+				WHEN @has_agent = 'disconnected' THEN
+					(
+						latest_build.disconnected_at IS NOT NULL AND
+						latest_build.disconnected_at > latest_build.last_connected_at
+					) OR (
+						latest_build.last_connected_at IS NOT NULL AND
+						latest_build.last_connected_at + INTERVAL '1 second' * @agent_inactive_disconnect_timeout :: bigint < NOW()
```

Comment:
Suggestion: Rename to `@agent_inactive_disconnect_timeout_seconds` to make the intent clear in the generated code.

Reply:
Good call, fixed.